### PR TITLE
Bluetooth: Mesh: Misc fixes

### PR DESCRIPTION
**Pull request by LingaoM**
This is a resubmission of PR (#30086)

Signed-off-by: Lingao Meng <menglingao@xiaomi.com>

### CODE REVIEW

**Comment by LingaoM** [comment]
@trond-snekvik Build CI looks like a problem?

**Comment by trond-snekvik** [comment]
Looks unrelated, could you try rebasing?

File: subsys/bluetooth/mesh/shell.c [hunk 1]
```diff
@@ -574,7 +574,7 @@ static int cmd_ident(const struct shell *shell, size_t argc, char *argv[])
 
 static int cmd_get_comp(const struct shell *shell, size_t argc, char *argv[])
 {
-	NET_BUF_SIMPLE_DEFINE(comp, 32);
+	NET_BUF_SIMPLE_DEFINE(comp, 64);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/shell.c:577]: Use `BT_MESH_RX_SDU_MAX`, so we can fix this once and for all

File: subsys/bluetooth/mesh/shell.c [hunk 2]
```diff
@@ -1705,7 +1705,6 @@ static int cmd_mod_pub(const struct shell *shell, size_t argc, char *argv[])
 	if (argc == 1 || argc == 8) {
 		cid = strtoul(argv[0], NULL, 0);
 		argc--;
-		argv++;
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/shell.c:1708]: What's the reasoning behind this change? The original looks correct to me

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/shell.c (hunk 2 above)
Comment (LingaoM) [reply subsys/bluetooth/mesh/shell.c:1708]: `argv += 3;` so now` argv[0] = pub.addr`, but if we continue increment by one, argv[0] will lost first params of publish set.

Comment (LingaoM) [reply subsys/bluetooth/mesh/shell.c:1708]: sorry, maybe i'm wrong.

**Review by trond-snekvik — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**